From f1c3a2d7b43d2a41e37c0bf4148114fe0ccfd483 Mon Sep 17 00:00:00 2001
From: mpvader <mvader@victronenergy.com>
Date: Tue, 19 May 2020 10:20:54 +0200
Subject: [PATCH] Leave CONFIG_ADVISE_SYSCALLS to its default (= enabled)

nodejs v10 needs madvise. And it was missing on te Beaglebone
due to above config, causing this error:

root@beaglebone:~# node
node[27753]: ../src/node_platform.cc:413:std::shared_ptr<node::PerIsolatePlatformData> node::NodePlatform::ForIsolate(v8::Isolate*): Assertion `data' failed.
Aborted

Other Venus OS kernels already include madvise.

v8 was fixed for this [1]. Which is not included in v10 [2], but is included in v11 [3].

[1] https://chromium.googlesource.com/v8/v8.git/+/ed0d279ed033a8e24d0a0a1869ff3cd41976797c
[2] https://github.com/nodejs/node/blob/v10.x/deps/v8/src/base/platform/platform-posix.cc#L159
[3] https://github.com/nodejs/node/blob/v11.x/deps/v8/src/base/platform/platform-posix.cc#L159
---
 arch/arm/configs/bbb_defconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm/configs/bbb_defconfig b/arch/arm/configs/bbb_defconfig
index f515802..47390af 100644
--- a/arch/arm/configs/bbb_defconfig
+++ b/arch/arm/configs/bbb_defconfig
@@ -15,7 +15,6 @@ CONFIG_BLK_DEV_INITRD=y
 # CONFIG_RD_LZO is not set
 # CONFIG_RD_LZ4 is not set
 CONFIG_KALLSYMS_ALL=y
-# CONFIG_ADVISE_SYSCALLS is not set
 CONFIG_EMBEDDED=y
 CONFIG_SLAB=y
 CONFIG_PROFILING=y
-- 
2.7.4

